### DISCUSSION TITLE
Diff against upstream branch

### DISCUSSION THREAD

**Discussion by dato** [jj-vcs/jj#4426] [Q&A]
Hello!

I'm a new user, and I find myself running this too often to know "what would push change":

```
$ jj diff --from push-xxxyyyzzz@origin --to @
```

(This is doing it manually for the current branch, and not everything that would be pushed.)

My questions are:

1. is there a shortcut/alias I could use for the command above? I read the revset docs, but I cannot figure out how to get the current tracking remote branch (sorry if it's documented somewhere, I couldn't find it)

2. could there perhaps be something like `jj git push --dry-run --diff` that shows me the changes that would be pushed (for all branches)?

Any tips welcome, thanks!

**Comment by bhearsum** [comment]
As an alternative, giving a revision change such that it can be fed to `jj log` easily would be helpful.

**Comment by arxanas** [comment]
There's a related discussion here, including a possible workaround: https://github.com/jj-vcs/jj/issues/4756
